src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -0,0 +1,1277 @@
+
+            // This should be the first write to our buffer. The trick of reserving space won't otherwise work.
+            Debug.Assert(_sendBuffer.ActiveLength == 0);
+
+            // Reserve space for header frame envelope.
+            _sendBuffer.Commit(PreHeadersReserveSpace);
+
+            // Add header block prefix. We aren't using dynamic table, so these are simple zeroes.
+            // https://tools.ietf.org/html/draft-ietf-quic-qpack-11#section-4.5.1
+            _sendBuffer.EnsureAvailableSpace(2);
+            _sendBuffer.AvailableSpan[0] = 0x00; // required insert count.
+            _sendBuffer.AvailableSpan[1] = 0x00; // s + delta base.
+            _sendBuffer.Commit(2);
+
+            HttpMethod normalizedMethod = HttpMethod.Normalize(request.Method);
+            if (normalizedMethod.Http3EncodedBytes != null)
+            {
+                BufferBytes(normalizedMethod.Http3EncodedBytes);
+            }
+            else
+            {
+                BufferLiteralHeaderWithStaticNameReference(H3StaticTable.MethodGet, normalizedMethod.Method);
+            }
+
+            BufferIndexedHeader(H3StaticTable.SchemeHttps);
+
+            if (request.HasHeaders && request.Headers.Host != null)
+            {
+                BufferLiteralHeaderWithStaticNameReference(H3StaticTable.Authority, request.Headers.Host);
+            }
+            else
+            {
+                BufferBytes(_connection.Pool._http3EncodedAuthorityHostHeader);
+            }
+
+            // The only way to reach H3 is to upgrade via an Alt-Svc header, so we can encode Alt-Used for every connection.
+            BufferBytes(_connection.AltUsedEncodedHeaderBytes);
+
+            string pathAndQuery = request.RequestUri.PathAndQuery;
+            if (pathAndQuery == "/")
+            {
+                BufferIndexedHeader(H3StaticTable.PathSlash);
+            }
+            else
+            {
+                BufferLiteralHeaderWithStaticNameReference(H3StaticTable.PathSlash, pathAndQuery);
+            }
+
+            if (request.HasHeaders)
+            {
+                // H3 does not support Transfer-Encoding: chunked.
+                request.Headers.TransferEncodingChunked = false;
+
+                BufferHeaderCollection(request.Headers);
+            }
+
+            if (_connection.Pool.Settings._useCookies)
+            {
+                string cookiesFromContainer = _connection.Pool.Settings._cookieContainer.GetCookieHeader(request.RequestUri);
+                if (cookiesFromContainer != string.Empty)
+                {
+                    BufferLiteralHeaderWithStaticNameReference(H3StaticTable.Cookie, cookiesFromContainer);
+                }
+            }
+
+            if (request.Content == null || request.Content.Headers.ContentLength == 0)
+            {
+                // Expect 100 Continue requires content.
+                request.Headers.ExpectContinue = null;
```

Comment:
We need to null this out?  If there's no content, seems we'd just ignore this, no?

Reply:
Nulled out so we don't send the header if the user set it incorrectly.

Reply:
Isn't this too late? Didn't you already send it on line 534?